### PR TITLE
manager-vrack-app

### PR DESCRIPTION
**Pull request by frenautvh**
# 🍕 Add vrack standalone module 🍕 
- commit history will be squashed when merging

## ⚠️ @ovh-ux/manager-cloud-styles
- dependency need to be updated after QC

## ⚠️ translations
- some already existing fr translations have been added to the module (translations comes from manager cloud)

### CODE REVIEW

File: packages/manager/modules/vrack/src/add/vrack-add.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                <div class="top-space-m20" data-ng-if="VrackAddCtrl.loaders.validationPending">
+                    <h2 class="text-center" data-translate="vrack_validationPending" data-translate-values="{url: VrackAddCtrl.model.purchaseOrderUrl}"></h2>
+                    <p class="text-center fs12" data-translate="vrack_order_delay"></p>
+                </div>
+
+                <!-- form -->
+                <div data-ng-if="!VrackAddCtrl.loaders.validationPending">
+                    <!-- contracts -->
+                    <div class="row" data-ng-if="!VrackAddCtrl.model.contractsAccepted && VrackAddCtrl.model.agreements.length"
+                         data-contracts="VrackAddCtrl.model.agreements"
```
Comment (antleblanc) [inline packages/manager/modules/vrack/src/add/vrack-add.html]: By using `contracts` directive it mean that you are linked to `cloud/client/components/contracts/contracts.directive.js`.

I'm not sure that we are providing a standalone module. Am I right?

File: packages/manager/modules/vrack/src/modals/selectVrack.controller.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import _ from 'lodash';
```
Comment (antleblanc) [inline packages/manager/modules/vrack/src/modals/selectVrack.controller.js]: You can import specific method from lodash4 if you want.

File: packages/manager/modules/vrack/src/vrack.component.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    $templateCache.put('vrack/partials/available.html', availablePartials);
+    $templateCache.put('vrack/partials/mapped.html', mappedPartials);
+  })
+  .service('VrackService', vrackService)
+  .component('ovhManagerVrackComponent', {
+    template,
+    controller,
+    controllerAs: 'VrackCtrl',
+  })
+  .config(($stateProvider) => {
```
Comment (antleblanc) [inline packages/manager/modules/vrack/src/vrack.component.js]:
```suggestion
  .config(/* @ngInject */($stateProvider) => {
```

File: packages/manager/modules/vrack/src/vrack.controller.js [hunk 4]
```diff
@@ -0,0 +1,735 @@
+import _ from 'lodash';
```
Comment (antleblanc) [inline packages/manager/modules/vrack/src/vrack.controller.js]: Could be removed if all needed methods are imported.

File: packages/manager/modules/vrack/src/vrack.service.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
+import _ from 'lodash';
+
+export default class VrackService {
+  /* @ ngInject */
```
Comment (antleblanc) [inline packages/manager/modules/vrack/src/vrack.service.js]:
```suggestion
  /* @ngInject */
```

File: packages/manager/modules/vrack/package.json [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  },
+  "devDependencies": {
+    "@ovh-ux/component-rollup-config": "^5.0.0-beta.9"
+  },
+  "peerDependencies": {
+    "@ovh-ux/manager-cloud-styles": "git+https://github.com/ovh-ux/manager.git#split-manager-cloud-styles/cloud-styles",
+    "@ovh-ux/ng-ovh-cloud-universe-components": "^0.1.0",
+    "@ovh-ux/manager-core": "^2.0.0",
+    "@uirouter/angularjs": "^1.0.15",
+    "angular": "1.7.5",
```
[the remark this answers is not in the document]
Comment (frenautvh) [reply packages/manager/modules/vrack/package.json]: Yes you're right, the problem right now is i'm using ovh-manager-core which has angular 1.7.5 as peer dependency. As suggested by @jleveugle maybe we should downgrade ovh-manager-core to angular 1.6 in order to preserve compatibility.
Migration from 1.6 to 1.7 seems tricky :/

**Comment by frenautvh** [comment]
clean rebase is needed

**Pull request closed without merging**